lib/preprocess.js

```diff
@@ -9,12 +10,27 @@ export const preprocess = async ({
   logger
 }) => {
   const measurements = (await fetchMeasurements(web3Storage, cid))
+    .map(measurement => {
+      try {
+        return {
+          ...measurement,
+          // eslint-disable-next-line camelcase
+          participantAddress: parseParticipantAddress(measurement.wallet_address)
```

Comment:
isn't this logic duplicated now, once in measure and once in filter? and filter still changes the measurement

Reply:
also, the measure service should submit `participant_address` instead of `wallet_address`. It's important that both services use the same terminology

Reply:
> isn't this logic duplicated now, once in measure and once in filter? and filter still changes the measurement

That was a mistake - I forgot to remove the code from the filter callback.

Fixed in 7c976f7e2e69cfaec821e2d45d6abe50fb255daa

> also, the measure service should submit `participant_address` instead of `wallet_address`. It's important that both services use the same terminology

In my view, SPARK modules knows nothing about participant addresses and FIL vs ETH formats. It receives the wallet address from the user and adds it to the measurement.

By keeping different field names (participant vs wallet), it's easier to reason about what kind of address is stored where.

- `wallet_address` is a Filecoin address (typically `f1...` or `f4...`)
- `participant_address` is an ETH address (`0x...`

Reply:
The problem is that you can't just send a filecoin address to the contract, if the argument is of type `address` (which `setScores` has). See the Slack discussion for a suggestion

Reply:
We decided against using FIL addresses, so that part of the contract can stay